File: Data Structures & Algorithms/house-robber-ii/submission-1.py
class Solution:
    def rob1(self, nums: List[int]) -> int:
        dp = [0] * (len(nums) + 2)

        for i, num in enumerate(nums):
            dp[i + 2] = max(dp[i + 1], dp[i] + num)

        return dp[-1]

    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
            
        return max(self.rob1(nums[1:]), self.rob1(nums[:-1]))

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.go
func longestConsecutive(nums []int) int {
	num_set := make(map[int]bool)

	for _, num := range nums {
		num_set[num] = true
	}

	ans := 0

	for num := range num_set {
		if _, exist := num_set[num-1]; exist {
			continue
		}

		cur := num
		leng := 1

		for num_set[cur+1] {
			cur += 1
			leng += 1
		}

		ans = max(ans, leng)
	}

	return ans
}

func max(a int, b int) int {
	if a > b {
		return a
	}

	return b
}